src/commands/kv/bucket.rs

```diff
@@ -0,0 +1,101 @@
+extern crate base64;
+
+use crate::commands::kv::delete_bulk::delete_bulk;
+use crate::commands::kv::write_bulk::write_bulk;
+
+use cloudflare::endpoints::workerskv::write_bulk::KeyValuePair;
+
+use walkdir::WalkDir;
+
+use std::ffi::OsString;
+use std::fs::metadata;
+use std::path::Path;
+
+use crate::terminal::message;
+
+pub fn upload(namespace_id: &str, filename: &Path) -> Result<(), failure::Error> {
+    let pairs: Result<Vec<KeyValuePair>, failure::Error> = match &metadata(filename) {
+        Ok(file_type) if file_type.is_dir() => parse_directory(filename),
+        Ok(_file_type) => {
+            // any other file types (files, symlinks)
+            failure::bail!("wrangler kv:bucket upload takes a directory")
+        }
+        Err(e) => failure::bail!("{}", e),
+    };
+
+    write_bulk(namespace_id, pairs?)
+}
+
+pub fn delete(namespace_id: &str, filename: &Path) -> Result<(), failure::Error> {
+    let keys: Result<Vec<String>, failure::Error> = match &metadata(filename) {
+        Ok(file_type) if file_type.is_dir() => {
+            let key_value_pairs = parse_directory(filename)?;
+            Ok(key_value_pairs
+                .iter()
+                .map(|kv| kv.key.clone())
+                .collect::<Vec<_>>())
```

Comment:
do we still think we can get rid of this cloning logic? I think it would be a good idea :)

Reply:
yes, apparently if it isn't in a PR comment I don't remember it.